flask/app/routes.py

```diff
@@ -16,7 +18,14 @@ def load_user(uid: int):
 @app.route('/api/login', methods=['POST'])
 def login():
     if current_user.is_authenticated:
-        return json.dumps({ "username": current_user.username }), 200
+        # get/update last login
+        last_login = current_user.last_login
+        try:
+            current_user.last_login = datetime.now()
+            db.session.commit()
+        except:
+            app.logger.info('failed to update last_login for %s', current_user.username)
+        return jsonify({ "username": current_user.username, "last_login": current_user.last_login }), 200
```

Comment:
Should return `last_login` or we're returning `datetime.now()` again! 
Also `jsonify` already returns a response object so we don't need the `, 200` in tuple

Reply:
resolved in 7e525b6